### PR TITLE
Delta source: add record count for DataFile

### PR DESCRIPTION
**Pull request by emilie-wang**
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

Addresses https://github.com/apache/incubator-xtable/issues/102

## Brief change log

-  Update how to get the number of records from Delta source 
- Side effect, it also fixed the edge case when columnStats list is empty, the number of records becomes 0

## Verify this pull request

This change added tests and can be verified as follows:

- added a line of test

### CODE REVIEW

File: xtable-core/src/main/java/org/apache/xtable/delta/DeltaStatsExtractor.java [hunk 1]
```diff
@@ -223,6 +223,15 @@ public List<ColumnStat> getColumnStatsForFile(AddFile addFile, List<InternalFiel
     }
   }
 
+  public long getNumRecords(AddFile addFile) {
```
Comment (the-other-tim-brown) [inline xtable-core/src/main/java/org/apache/xtable/delta/DeltaStatsExtractor.java:226]: Instead of parsing the stats twice, is there a way we can include this with the column stats output of `getColumnStatsForFile`? Maybe we can have some wrapper that wraps the count with the formatted stats?

Comment (emilie-wang) [reply xtable-core/src/main/java/org/apache/xtable/delta/DeltaStatsExtractor.java]: Sure, I created a wrapper called `FileStats`, which contains `List<ColumnStat>` and `numRecords` for per file.

**Review by the-other-tim-brown — APPROVED** [review]
LGTM! Can you squash down to a single commit for me? After that, I will merge the PR.

**Comment by emilie-wang** [comment]
@the-other-tim-brown Thank you for the review and I have squashed down to 1 commit.
